storage/core/src/main/java/com/hortonworks/registries/storage/impl/jdbc/JdbcStorageManager.java

```diff
@@ -61,29 +62,96 @@ public JdbcStorageManager(QueryExecutor queryExecutor) {
 
     @Override
     public void add(Storable storable) throws AlreadyExistsException {
-        log.debug("Adding storable [{}]", storable);
-        queryExecutor.insert(storable);
+        boolean committed = false;
+        try {
+            beginTransaction(TransactionIsolation.DEFAULT);
```

Comment:
the transactions are handled at a higher level via [ManagedTransaction](https://github.com/hortonworks/registry/blob/master/storage/core/src/main/java/com/hortonworks/registries/storage/transaction/ManagedTransaction.java).

I dont think the Storage Manager should support txn for individual operations. There could be multiple storage manager operations that needs to be part of a txn.